eng/TestRunsheetBuilder/TestRunsheetBuilder.targets

```diff
@@ -0,0 +1,109 @@
+<Project>
+  <!--
+
+    This file is used to generate a list of tests to run on GitHub Actions.
+
+    - For projects not requiring packages:
+         .\build.cmd -test /p:TestRunnerName=TestRunsheetBuilder [/bl /p:GITHUB_ACTIONS=true]
+
+    - For projects requiring packages (e.g., E2E, templates and playground):
+         .\build.cmd -test /p:TestRunnerName=TestRunsheetBuilder /p:FullE2e=true [/bl /p:GITHUB_ACTIONS=true]
+
+    For the large part this is a copy of the Arcade SDK's implementations:
+      - https://github.com/dotnet/arcade/blob/b888df17/src/Microsoft.DotNet.Arcade.Sdk/tools/XUnit/XUnit.Runner.targets
+      - https://github.com/dotnet/arcade/blob/b888df17/src/Microsoft.DotNet.Arcade.Sdk/tools/VSTest.targets
+   -->
+
+  <Target Name="RunTests"
+          Outputs="%(TestToRun.ResultsStdOutPath)"
+          Condition=" '$(SkipTests)' != 'true' and '$(IsGitHubActionsRunner)' == 'true' and '$(RunOnGithubActions)' == 'true' ">
+
+    <PropertyGroup>
+      <!--
+
+        The runsheet is created in two scenarios:
+        - For projects requiring packages, only if full end-to-end tests (`FullE2e`) are enabled.
+        - For projects not requiring packages, only if full end-to-end tests (`FullE2e`) are disabled.
+
+        This logic ensures runsheets are generated appropriately based on the project's characteristics and the testing scenario.
+
+        FIXME: Aspire.Playground.Tests are currently not running in "out of repo" mode.
+        FIXME: Aspire.Templates.Tests are currently left alone.
+
+        -->
+      <_RequiresPackages Condition=" '$(MSBuildProjectName)' == 'Aspire.EndToEnd.Tests' ">true</_RequiresPackages>
+
+      <_CreateRunsheet>false</_CreateRunsheet>
+      <_CreateRunsheet Condition=" '$(_RequiresPackages)' == 'true' and '$(FullE2e)' == 'true' ">true</_CreateRunsheet>
+      <_CreateRunsheet Condition=" '$(_RequiresPackages)' != 'true' and '$(FullE2e)' != 'true' ">true</_CreateRunsheet>
+
+      <_CreateRunsheet Condition=" '$(MSBuildProjectName)' == 'Aspire.Templates.Tests' ">false</_CreateRunsheet>
```

Comment:
@radical this is where the Templates are locked out atm; it should be a matter of removing this line.